curator-recipes/src/main/java/org/apache/curator/framework/recipes/cache/TreeCache.java

```diff
@@ -758,6 +758,7 @@ private void handleStateChange(ConnectionState newState)
             break;
 
         case LOST:
```

Comment:
LOST doesn't mean session lost. It only means that the retry policy has given up. Is this your intent?